Tests/GitHubTeams.tests.ps1

```diff
@@ -11,13 +11,516 @@
     Justification='Suppress false positives in Pester code blocks')]
 param()
 
+Set-StrictMode -Version 1.0
+
 # This is common test code setup logic for all Pester test files
 $moduleRootPath = Split-Path -Path $PSScriptRoot -Parent
 . (Join-Path -Path $moduleRootPath -ChildPath 'Tests\Common.ps1')
 
 try
 {
-    # TODO once more capabilities exist in the module's API set
+    # Define Script-scoped, readonly, hidden variables.
+    @{
+        defaultRepoDesc = "This is a description."
+        defaultRepoHomePage = "https://www.microsoft.com/"
+        defaultRepoTopic = "microsoft"
+    }.GetEnumerator() | ForEach-Object {
+        Set-Variable -Force -Scope Script -Option ReadOnly -Visibility Private -Name $_.Key -Value $_.Value
+    }
+
+    Describe 'GitHubTeams\Get-GitHubTeam' {
+
+        Context 'When getting a GitHub Team by organization' {
+            BeforeAll {
+                $repoName = [Guid]::NewGuid().Guid
+
+                $newGithubRepositoryParms = @{
+                    RepositoryName = $repoName
+                    OrganizationName = $script:organizationName
+                }
+                $repo = New-GitHubRepository @newGitHubRepositoryParms
+
+                $teamName = [Guid]::NewGuid().Guid
+                $description = 'Team Description'
+                $privacy = 'closed'
+                $maintainers = $script:ownerName
+
+                $newGithubTeamParms = @{
+                    OrganizationName = $script:organizationName
+                    TeamName = $teamName
+                    Description = $description
+                    RepositoryName = $repo.full_name
+                    Privacy = $privacy
+                    Maintainers = $maintainers
+                }
+                New-GitHubTeam @newGithubTeamParms | Out-Null
+
+                $getGitHubTeamParms = @{
+                    OrganizationName = $script:organizationName
+                    TeamName = $teamName
+                }
+                $team = Get-GitHubTeam @getGitHubTeamParms
+            }
+
+            It 'Should return an object of the correct type' {
+                $team | Should -BeOfType PSCustomObject
+            }
+
+            It 'Should return the correct properties' {
+                $team.name | Should -Be $teamName
+                $team.description | Should -Be $description
+                $team.organization.login | Should -Be $script:organizationName
+                $team.parent | Should -BeNullOrEmpty
+                $team.members_count | Should -Be 1
+                $team.repos_count | Should -Be 1
+                $team.privacy | Should -Be $privacy
+            }
+
+            AfterAll {
+                if ($team)
```

Comment:
It's not clear to me that this conditional is needed.  I believe that AfterAll is going to run even if the internal tests fail.  And if BeforeAll fails, either AfterAll will also fail and error, or it won't run at all.  Either way, I'm not sure it adds much additional value, plus this will likely be able to be simplified to just `$team | Remove-GitHubTeam -Force` once pipelining is done.  Simple, easy to read one-liner.

Here and elsewhere (except for the Remove-GitHubTeam tests themselves of course)